tools/CheckMetadata.py

```diff
@@ -0,0 +1,189 @@
+        self.assertEqual(publication_date.day, int(version_day))
+
+    def test_doi(self):
+        self.assertIn('Doi', self.metadata)
+        doi = self.metadata['Doi']
+        self.assertRegex(doi, '^' + DOI_PATTERN + '$')
+
+    def test_zenodo_id(self):
+        self.assertIn('ZenodoId', self.metadata)
+        zenodo_id = self.metadata['ZenodoId']
+        self.assertTrue(isinstance(zenodo_id, int))
+        zenodo_id_from_doi = re.match(DOI_PATTERN,
+                                      self.metadata['Doi']).group(1)
+        self.assertEqual(zenodo_id, int(zenodo_id_from_doi),
+                         "Zenodo ID should match DOI")
+
+    def test_description(self):
+        self.assertIn('Description', self.metadata)
+        self.assertTrue(isinstance(self.metadata['Description'], str))
+
+    def test_keywords(self):
+        self.assertIn('Keywords', self.metadata)
+        keywords = self.metadata['Keywords']
+        self.assertTrue(isinstance(keywords, list))
+        self.assertTrue([isinstance(keyword, str) for keyword in keywords])
+
+    def test_affiliations(self):
+        self.assertIn('Affiliations', self.metadata)
+        affiliations = self.metadata['Affiliations']
+        self.assertTrue(isinstance(affiliations, list))
+        self.assertTrue(
+            [isinstance(affiliation, str) for affiliation in affiliations])
+        # Check for duplicates
+        simplified_affiliations = [
+            simplify_name(affiliation) for affiliation in affiliations
+        ]
+        for affiliation in affiliations:
+            self.assertEqual(
+                simplified_affiliations.count(simplify_name(affiliation)), 1,
+                f"Duplicate affiliation: {affiliation}")
+
+    def test_authors(self):
+        self.assertIn('Authors', self.metadata)
+        authors = self.metadata['Authors']
+        self.assertTrue(isinstance(authors['Description'], str))
+        authors_core = authors['Core']
+        self.assertTrue(isinstance(authors_core['Description'], str))
+        authors_core_list = authors_core['List']
+        self.assertTrue(isinstance(authors_core_list, list))
+        authors_devs = authors['Developers']
+        self.assertTrue(isinstance(authors_devs['Description'], str))
+        authors_devs_list = authors_devs['List']
+        self.assertTrue(isinstance(authors_devs_list, list))
+        authors_contribs = authors['Contributors']
+        self.assertTrue(isinstance(authors_contribs['Description'], str))
+        authors_contribs_list = authors_contribs['List']
+        self.assertTrue(isinstance(authors_contribs_list, list))
+        all_authors = (authors_core_list + authors_devs_list +
+                       authors_contribs_list)
+
+        # Check alphabetical ordering
+        self.assertEqual(
+            sorted(authors_devs_list,
+                   key=lambda a: a['Name']), authors_devs_list,
+            "'Developers' author list should be ordered alphabetically")
```

Comment:
Is there a good way to list which author(s) are not in order?

Also, `alphabetically by last name` (just to be extra clear)

Same below

Reply:
The `self.assertEqual` prints that information by default:

```
======================================================================
FAIL: test_authors (__main__.TestMetadata)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tools/CheckMetadata.py", line 136, in test_authors
    self.assertEqual(
AssertionError: Lists differ: [{'Na[1762 chars]e': 'Scheel, Mark A.', 'Orcid': '0000-0001-665[531 chars]ky'}] != [{'Na[1762 chars]e': 'Teukolsky, Saul A.', 'Orcid': '0000-0001-[531 chars]el'}]

First differing element 8:
{'Name': 'Scheel, Mark A.', 'Orcid': '0000-0001-665[182 chars]eel'}
{'Name': 'Teukolsky, Saul A.', 'Orcid': '0000-0001-[290 chars]sky'}

Diff is 3340 characters long. Set self.maxDiff to None to see it. : 'Developers' author list should be ordered alphabetically by last name
```